specs/windows/windows_search.table

```diff
@@ -0,0 +1,13 @@
+table_name("windows_search")
```

Comment:
what if schema is simplified a bit and just a fixed number of [item](https://learn.microsoft.com/en-us/windows/win32/properties/windows-properties-system) [properties](https://learn.microsoft.com/en-us/windows/win32/properties/props-system-itempathdisplay) are being shown? 

There is a distinction here between the properties you put on the query to narrow the search and the item properties displayed here. You can match for `txt` files that have the word hello on them and don't necessarily have to show the file extension of the matched items on the results.

This is a suggestion and potentially a way to simplify things. Let me know what you think.

```
description("Run searches against the windows system index database.")
schema([
    Column("name", TEXT, "The name of the item"),
    Column("path", TEXT, "The full path of the item."),
    Column("size", BIGINT, "The item size in bytes."),
    Column("date_created", TEXT, " The date and time the item was created."),
    Column("date_modified", TEXT, "The date and time the item was last modified"),
    Column("attributes", TEXT, "The item attributes"),
    Column("owner", TEXT, "The owner of the item"),
    Column("type", TEXT, "The item type"),
    Column("extra_properties", TEXT, "Placeholder for type-specific properties"), // System.Title, System.Author, System.Keywords, etc
    Column("query", TEXT, "windows search query", required=True),
])
```

Reply:
@James-Pickett Thanks for looking into my suggestion! The current schema returns more than one row per search hit. I think it would be easier if only one row is returned per hit. I'd not include max_results, sort, or attribute in this scenario.

@directionless, you have a more experience with table schemas. What is your advice on scenarios like this one where the returned data can be represented in more than one row? I'd prefer to have a schema that can handle one row per search hit

description("Run searches against the windows system index database.")
schema([
    Column("name", TEXT, "The name of the item"),
    Column("path", TEXT, "The full path of the item."),
    Column("size", BIGINT, "The item size in bytes."),
    Column("date_created", TEXT, " The date and time the item was created."),
    Column("date_modified", TEXT, "The date and time the item was last modified"),
    Column("attributes", TEXT, "The item attributes"),
    Column("owner", TEXT, "The owner of the item"),
    Column("type", TEXT, "The item type"),
    Column("locale", BIGINT, "The item locale"),    
    Column("extra_properties", TEXT, "Placeholder for type-specific properties"), // System.Title, System.Author, System.Keywords, etc
    Column("query", TEXT, "windows search query", required=True),
])

Reply:
Talking a little about this in office hours, we talked about whether to make sense to use full EAV, or pick some common things, and pack extra things into a json structure in `extra_properties`

My initial bias is for EAV, but mostly because I'm used to it. But I could imagine either way.

Reply:
thought about this some more and watched the last office hours, working on including the columns you suggested as well as a json of "additional_properties", here is a sample. I think it's the best of both worlds, can query for what ever you want, but get sane default columns.

```sql
osquery> select * from windows_search where sort = 'system.size desc' and max_results = '4' and additional_properties = 'system.itemurl';
+--------------+-------------------------------------------------------------------------------------------------------------------------------------------------------+------------+--------------+---------------+-------+-------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| name         | path                                                                                                                                                  | size       | date_created | date_modified | owner | type  | properties                                                                                                                                                                      |
+--------------+-------------------------------------------------------------------------------------------------------------------------------------------------------+------------+--------------+---------------+-------+-------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| usmt.ppkg    | C:\Recovery\Customizations\usmt.ppkg                                                                                                                  | 3116066935 | 1670959177   | 1670959754    | NULL  | .ppkg | {"system.itemurl":"file:C:/Recovery/Customizations/usmt.ppkg"}                                                                                                                  |
| payload.vsix | C:\ProgramData\Microsoft\VisualStudio\Packages\Microsoft.VisualStudio.VC.Llvm.Clang,version=16.11.31503.43\payload.vsix                               | 1023028086 | 1678829483   | 1678829441    | NULL  | .vsix | {"system.itemurl":"file:C:/ProgramData/Microsoft/VisualStudio/Packages/Microsoft.VisualStudio.VC.Llvm.Clang,version=16.11.31503.43/payload.vsix"}                               |
| content.bin  | C:\Windows\ServiceProfiles\NetworkService\AppData\Local\Microsoft\Windows\DeliveryOptimization\Cache\1004e9b0-b5c9-477a-a13b-a67f4fe2a0c7\content.bin | 426180978  | 1683657185   | 1683657206    | NULL  | .bin  | {"system.itemurl":"file:C:/Windows/ServiceProfiles/NetworkService/AppData/Local/Microsoft/Windows/DeliveryOptimization/Cache/1004e9b0-b5c9-477a-a13b-a67f4fe2a0c7/content.bin"} |
| chrome.7z    | C:\Program Files\BraveSoftware\Brave-Browser\Application\113.1.51.114\Installer\chrome.7z                                                             | 336465005  | 1683695947   | 1683628208    | NULL  | .7z   | {"system.itemurl":"file:C:/Program Files/BraveSoftware/Brave-Browser/Application/113.1.51.114/Installer/chrome.7z"}                                                             |
+--------------+-------------------------------------------------------------------------------------------------------------------------------------------------------+------------+--------------+---------------+-------+-------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```